cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java

```diff
@@ -380,11 +384,25 @@ public Cancellable announce(String serviceName, int port) {
           });
 
           bind(Store.class).to(DefaultStore.class);
+
+          // For binding StreamWriter
+          install(createStreamFactoryModule());
         }
       }
     );
   }
 
+  private Module createStreamFactoryModule() {
+    return new PrivateModule() {
+      @Override
+      protected void configure() {
+        install(new FactoryModuleBuilder().implement(StreamWriter.class, DefaultStreamWriter.class)
+                  .build(StreamWriterFactory.class));
+        expose(StreamWriterFactory.class);
```

Comment:
If you only have one binding and is always exposed, you don't need to use PrivateModule. Just using AbstractModule is fine.